test/powershell/engine/Api/GetMethodInfoOverloadDefinition.Tests.ps1

```diff
@@ -0,0 +1,192 @@
+            @{
+                Description = "optional string parameter";
+                MethodName = "TestMethod_OptString";
+                ExpectedDefinition = 'static void TestMethod_OptString(string optParam = "test string")'
+            }
+            @{
+                Description = "optional struct parameter";
+                MethodName = "TestMethod_OptStruct";
+                ExpectedDefinition = 'static void TestMethod_OptStruct(datetime optParam = default)'
+            }
+            @{
+                Description = "optional enum parameter";
+                MethodName = "TestMethod_OptEnum";
+                ExpectedDefinition = "static void TestMethod_OptEnum(System.UriKind optParam = System.UriKind.Relative)"
+            }
+            @{
+                Description = "optional generic parameter";
+                MethodName = "TestMethod_OptGeneric";
+                ExpectedDefinition = "static void TestMethod_OptGeneric[T](T param = default)"
+            }
+        )
+    }
+
+    Context "Verify method definitions" {
+
+        It "Get methods' overload definitions" {
+            $definitions = ([OverloadDefinitionTest.TestClass]::Bar).OverloadDefinitions
+            $definitions.Count | Should -Be 4
+            $definitions[0] | Should -BeExactly "static void Bar()"
+            $definitions[1] | Should -BeExactly "static void Bar(int param)"
+            $definitions[2] | Should -BeExactly "static void Bar(int param1, string param2)"
+            $definitions[3] | Should -BeExactly "static string Bar(System.Nullable[int] param1, string param2)"
+        }
+
+        It "Get definition of the method with <Description>" -TestCases $testCases {
+            param($MethodName, $ExpectedDefinition)
+
+            $definition = ([MethodDefinitionTest.TestClass] | Get-Member -Type Method -Static $MethodName).Definition
+            $definition | Should -BeExactly $ExpectedDefinition
+
+            $overloadDefinition = ([MethodDefinitionTest.TestClass]::$MethodName).OverloadDefinitions
+            $overloadDefinition | Should -BeExactly $definition
+        }
+    }
+
+    Context "Verify method definitions' tooltip" {
+
+        It "Tooltip should contain methods' overload definitions" {
+            $result = (TabExpansion2 -inputScript ($s = '[OverloadDefinitionTest.TestClass]::Bar') -cursorColumn $s.Length).CompletionMatches
+            $tooltipParts = $result.ToolTip -split "\r?\n"
+
+            $tooltipParts.Count | Should -Be 4
+            $tooltipParts[0] | Should -BeExactly "static void Bar()"
+            $tooltipParts[1] | Should -BeExactly "static void Bar(int param)"
+            $tooltipParts[2] | Should -BeExactly "static void Bar(int param1, string param2)"
+            $tooltipParts[3] | Should -BeExactly "static string Bar(System.Nullable[int] param1, string param2)"
+        }
+
+        It "Tooltip should contain definition of the method with <Description>" -TestCases $testCases {
```

Comment:
Do we really need the testCase (so many tests)?

Reply:
I've used `testCases` to verify tooltip for the different method signatures with optional parameters and to avoid duplicating code. If you think using all of the test cases is redundant I'll edit the test.